### ISSUE TITLE
Fails when compiled with esp-idf-4.3.1

### ISSUE THREAD

**Issue by wibbler** [nopnop2002/esp-idf-a2dp-source#2]
This seems to fail when compiled with esp-idf v4.3.1

**Comment by nopnop2002** [comment]
Try this:
```
git clone https://github.com/nopnop2002/esp-idf-a2dp-source
cd esp-idf-a2dp-source
idf.py set-target esp32
idf.py menuconfig
idf.py flash monitor
```

If it fails, please update esp-idf.

**Comment by wibbler** [comment]
Thanks,
Tried with the set-target clause, didn't use that before. now plays for a second or so then stops, esp-idf is latest stable version (4.3.1).
console log: (with the E obviously in red).

Tried with 4.1.1 worked fine, but needed to comment out ESP_BT_GAP_MODE_CHG_EVT in main.c.

I (15831) RCCT: remote rn_cap: count 1, bitmask 0x2000
I (21191) BT_AV: ESP_BT_GAP_MODE_CHG_EVT mode:2
I (21421) BT_AV: bt_app_av_sm_hdlr state 5, evt 0xff00
I (21421) BT_AV: a2dp media ready checking ...
I (21421) BT_AV: bt_app_av_sm_hdlr state 5, evt 0x3
I (21421) BT_AV: a2dp media ready, starting ...
W (21431) BT_APPL: new conn_srvc id:18, app_id:1
I (21861) BT_AV: ESP_BT_GAP_MODE_CHG_EVT mode:0
I (21891) BT_LOG: bta_av_link_role_ok hndl:x41 role:0 conn_audio:x1 bits:1 features:x824b

I (21891) BT_AV: bt_app_av_sm_hdlr state 5, evt 0x3
I (21891) BT_AV: a2dp media start successfully.
I (21901) BT_AV: bt_app_av_sm_hdlr state 5, evt 0x1
E (21931) BT_L2CAP: l2cab is_cong_cback_context
E (21941) BT_L2CAP: l2cab is_cong_cback_context
E (22021) BT_L2CAP: l2cab is_cong_cback_context
E (22031) BT_L2CAP: l2cab is_cong_cback_context
E (22051) BT_L2CAP: l2cab is_cong_cback_context
E (22061) BT_L2CAP: l2cab is_cong_cback_context
E (22061) BT_L2CAP: l2cab is_cong_cback_context
E (22081) BT_L2CAP: l2cab is_cong_cback_context

**Comment by nopnop2002** [comment]
I will look it up.

**Comment by nopnop2002** [comment]
I found that esp_bt_gap_cancel_discovery function does not work properly on ESP-IDF V4.4.


I reported here.

https://github.com/espressif/esp-idf/issues/7561

**Comment by nopnop2002** [comment]
It's fixed with ESP-IDF version:v4.4-dev-3573-gfb24a2941c-dirty or later.
